src/vxworks.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation for VxWorks
+use crate::error::{Error, RAND_SECURE_FATAL};
+use crate::util_libc::last_os_error;
+use core::sync::atomic::{AtomicBool, Ordering::Relaxed};
+
+static RNG_INIT: AtomicBool = AtomicBool::new(false);
+
+pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
+    while !RNG_INIT.load(Relaxed) {
```

Comment:
We should use `LazyBool` here, something like:

```rust
fn rng_init() -> bool {
    loop {
        let ret = unsafe { libc::randSecure() }
        if ret != 0 {
            return ret > 0;
        }
        unsafe { libc::usleep(10) };
    }
}
```

```rust
pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
    static RNG_INIT: LazyBool = LazyBool::new();
    if !RNG_INIT.unsync_init(rng_init) {
        return Err(RAND_SECURE_FATAL);
    }

    // Prevent overflow of i32
    ...
```

You could also use a lambda, if you think it would look better.

Reply:
@BaoshanPang @n-salim
Is it possible for the system to recover after `randSecure` failure or can we assume that once it has failed it will always fail?

Reply:
Yes, the system can recover after randSecure failure. This is the randSecure's function desciption:

* randSecure - determine if state is secure                                                                                        
*                                                                                                                                  
* This routine determines if state is secure and returns status.                                                                   
* When the system boots, the SW random number lib contains no entropy. Once                                                        
* enough entropy has been collected in order to make the random numbers                                                            
* cryptographically secure, the state is considered to be secure.                                                                  
*                                                                                                                                  
* It will make a system call to a kernel side handler.                                                                             
*                                                                                                                                  
* RETURNS: 1 if the state is secure, 0 if the state is not secure, ERROR if                                                        
* the random number generator module is not initialized.                                                                           
*                                                                                                                                  
* ERRNOS: N/A

Reply:
> "ERROR if the random number generator module is not initialized."

If I'm understanding this right, the RNG module might not be initialized, but it could become initialized later?

If that's the case, the current code looks good to me.